### PR TITLE
[Core] Test Framework: request body validation skipped when not recording

### PR DESCRIPTION
**Pull request by kinelski**
Skipping request body validation for requests sent in a `DontRecord` or a `RecordWithoutRequestBody` scope.

### CODE REVIEW

File: sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public RecordMatcher(bool compareBodies = true)
             "x-ms-correlation-request-id"
         };
 
-        public virtual RecordEntry FindMatch(RecordEntry request, IList<RecordEntry> entries)
+        public virtual RecordEntry FindMatch(RecordEntry request, IList<RecordEntry> entries, bool skipRequestBody)
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs:61]: Why does FindMatch has to flow up to here?

Can we just clean request.Body before passing it in like we do in RecordingTransport before storing?

File: sdk/core/Azure.Core.TestFramework/src/RecordSession.cs [hunk 2]
```diff
@@ -79,14 +79,16 @@ public void Record(RecordEntry entry)
             }
         }
 
-        public RecordEntry Lookup(Request request, RecordMatcher matcher, RecordedTestSanitizer sanitizer)
+        public RecordEntry Lookup(Request request, RecordMatcher matcher, RecordedTestSanitizer sanitizer, Func<RecordEntry, bool> skipRequestBodyFilter)
         {
             var requestEntry = RecordTransport.CreateEntry(request, null);
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/RecordSession.cs:84]: Let's pull `requestEntry` creating into PlaybackTransport.cs and use the filter to reset body there.

This way we can limit the spread of this flag just to the transports.

File: sdk/core/Azure.Core.TestFramework/src/TestRecording.cs [hunk 3]
```diff
@@ -201,7 +201,8 @@ public HttpPipelineTransport CreateTransport(HttpPipelineTransport currentTransp
             {
                 RecordedTestMode.Live => currentTransport,
                 RecordedTestMode.Record => new RecordTransport(_session, currentTransport, entry => _disableRecording.Value, Random),
-                RecordedTestMode.Playback => new PlaybackTransport(_session, _matcher, _sanitizer, Random),
+                RecordedTestMode.Playback => new PlaybackTransport(_session, _matcher, _sanitizer, Random,
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/TestRecording.cs:204]: `EntryRecordModel.DontRecord` in theory should never be set when sending a request in playback mode.

We don't have entry to match it with anyway.

Comment (kinelski) [reply sdk/core/Azure.Core.TestFramework/src/TestRecording.cs:204]: Good point.

**Review by pakrym — APPROVED** [review]

**Pull request merged**